### PR TITLE
Add timer trigger for status and notification sending

### PR DESCRIPTION
**Pull request by FirestarJes**

### CODE REVIEW

File: source/settlement-report/DatabaseMigration/Scripts/202410261405_Add_IsNotificationSent_to_SettlementReport.sql [hunk 1]
```diff
@@ -0,0 +1,3 @@
+ALTER TABLE [settlementreports].[SettlementReport]
+ADD [IsNotficationSent] [bit] NOT NULL DEFAULT(1); --For all existing records, we will assume that notification has been sent.
```
Comment (defectiveAi) [inline source/settlement-report/DatabaseMigration/Scripts/202410261405_Add_IsNotificationSent_to_SettlementReport.sql:2]: Spelling of column.

File: source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/DispatchIntegrationEventsTrigger.cs [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    public DispatchIntegrationEventsTrigger(IPublisher publisher)
+    {
+        _publisher = publisher;
+    }
+
+    [Function(nameof(DispatchIntegrationEventsTrigger))]
+    public Task RunAsync([TimerTrigger("* */1 * * *")] FunctionContext context)
+    {
+        ArgumentNullException.ThrowIfNull(context);
+        return _publisher.PublishAsync(context.CancellationToken);
```
Comment (defectiveAi) [inline source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/DispatchIntegrationEventsTrigger.cs:33]: Move this to the end of the other trigger, so we handle status, then send events one after the other.

File: source/settlement-report/SettlementReports.Application/SettlementReports_v2/SettlementReport.cs [hunk 3]
```diff
@@ -58,6 +58,8 @@ public sealed class SettlementReport
 
     public long? JobId { get; init; }
 
+    public bool IsNotficationSent { get; private set; }
```
Comment (defectiveAi) [inline source/settlement-report/SettlementReports.Application/SettlementReports_v2/SettlementReport.cs:61]: Spelling

File: source/settlement-report/SettlementReports.Infrastructure/Notifications/IntegrationEventProvider.cs [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+                {
+                    SettlementReportStatus.Completed => "SettlementReportReadyForDownload",
+                    SettlementReportStatus.Failed => "SettlementReportFailed",
+                    _ => throw new InvalidOperationException("Sending notification for settlement report with status other than Completed or Failed is not supported"),
+                },
+                TargetActorId = reportForNotification.ActorId.ToString(),
+                TargetUserId = reportForNotification.UserId.ToString(),
+                RelatedId = reportForNotification.Id.ToString(),
+                OccurredAt = now.ToTimestamp(),
+                ExpiresAt = now.AddHours(23).ToTimestamp(),
```
Comment (defectiveAi) [inline source/settlement-report/SettlementReports.Infrastructure/Notifications/IntegrationEventProvider.cs:71]: Maybe 6 days? Reports are deleted at 7th day.

File: source/settlement-report/DatabaseMigration/Scripts/202410261405_Add_IsNotificationSent_to_SettlementReport.sql (hunk 1 above)
Comment (FirestarJes) [reply source/settlement-report/DatabaseMigration/Scripts/202410261405_Add_IsNotificationSent_to_SettlementReport.sql:2]: Doh :P

File: source/settlement-report/SettlementReports.Infrastructure/Notifications/IntegrationEventProvider.cs (hunk 4 above)
Comment (FirestarJes) [reply source/settlement-report/SettlementReports.Infrastructure/Notifications/IntegrationEventProvider.cs:71]: ok

Comment (FirestarJes) [reply source/settlement-report/SettlementReports.Infrastructure/Notifications/IntegrationEventProvider.cs:71]: Fixed

File: source/settlement-report/SettlementReports.Application/SettlementReports_v2/SettlementReport.cs (hunk 3 above)
Comment (FirestarJes) [reply source/settlement-report/SettlementReports.Application/SettlementReports_v2/SettlementReport.cs:61]: Fixed

File: source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/DispatchIntegrationEventsTrigger.cs (hunk 2 above)
Comment (FirestarJes) [reply source/settlement-report/Orchestration.SettlementReports/Functions/SettlementReports/DispatchIntegrationEventsTrigger.cs:33]: Fixed

File: source/settlement-report/DatabaseMigration/Scripts/202410261405_Add_IsNotificationSent_to_SettlementReport.sql (hunk 1 above)
Comment (FirestarJes) [reply source/settlement-report/DatabaseMigration/Scripts/202410261405_Add_IsNotificationSent_to_SettlementReport.sql:2]: fixed

File: source/settlement-report/SettlementReports.Application/SettlementReports_v2/ISettlementReportRepository.cs [hunk 5]
```diff
@@ -31,4 +31,6 @@ public interface ISettlementReportRepository
     Task<IEnumerable<SettlementReport>> GetForJobsAsync();
 
     Task<IEnumerable<SettlementReport>> GetForJobsAsync(Guid actorId);
+
+    Task<IEnumerable<SettlementReport>> GetNeedsNotificationSentForCompletedAndFailed();
```
Comment (defectiveAi) [inline source/settlement-report/SettlementReports.Application/SettlementReports_v2/ISettlementReportRepository.cs:35]: nitpick: GetPendingNotificationsForCompletedAndFailed or something like that.

File: source/settlement-report/SettlementReports.Function/Program.cs [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+using Energinet.DataHub.SettlementReport.Common.Infrastructure.Security;
+using Energinet.DataHub.SettlementReport.Common.Infrastructure.Telemetry;
+using Energinet.DataHub.SettlementReport.Infrastructure.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+var host = new HostBuilder()
+    .ConfigureFunctionsWebApplication(worker =>
+    {
+        // Http => Authentication
+        worker.UseUserMiddlewareForIsolatedWorker<FrontendUser>();
```
Comment (defectiveAi) [inline source/settlement-report/SettlementReports.Function/Program.cs:28]: Remove auth registration, is not needed.

Comment (FirestarJes) [reply source/settlement-report/SettlementReports.Function/Program.cs:28]: fixed

File: source/settlement-report/SettlementReports.Application/SettlementReports_v2/ISettlementReportRepository.cs (hunk 5 above)
Comment (FirestarJes) [reply source/settlement-report/SettlementReports.Application/SettlementReports_v2/ISettlementReportRepository.cs]: Renamed

**Review by defectiveAi — APPROVED** [review]

**Review by lanni-energinet — APPROVED** [review]

**Pull request merged**